### DISCUSSION TITLE
How to extend the APIs?

### DISCUSSION THREAD

**Discussion by afshinm** [apache/superset#14376] [Q&A / Help]
I'm currently using BigQuery as the DB backend but I'd like to extend the API and some additional features, e.g. `dryRun` feature of BigQuery to get a resource/time estimate of the query before actually running it. 

My question is, how can I extend the API side without deeply changing the core? In other words, can I write a plugin for Flask/Blueprint to add my endpoint?

(I also understand my question might be a bit vague, sorry).

**Comment by srinify** [comment]
@betodealmeida I think you've mentioned query cost estimation for Presto and maybe BigQuery too? Any pointers here
